spatialprofilingtoolbox/cggnn/scripts/run.py

```diff
@@ -120,63 +117,21 @@ def parse_arguments():
     return parser.parse_args()
 
 
-def _create_cell_df(cell_dfs: dict[str, DataFrame], feature_names: dict[str, str]) -> DataFrame:
-    "Find chemical species, phenotypes, and locations and merge into a DataFrame."
-
-    for specimen, df_specimen in cell_dfs.items():
-        df_specimen.rename({ft_id: 'FT_' + ft_name for ft_id, ft_name in feature_names.items()},
-                           axis=1, inplace=True)
-        # TODO: Create phenotype columns
-        df_specimen.rename({'pixel x': 'center_x', 'pixel y': 'center_y'},
-                           axis=1, inplace=True)
-        df_specimen['specimen'] = specimen
-
-    # TODO: Reorder so that it's feature, specimen, phenotype, xy
-    # TODO: Verify histological structure ID or recreate one
-    df = concat(cell_dfs.values(), axis=0)
-    df.index.name = 'histological_structure'
-    return df
-
-
-def _create_label_df(df_assignments: DataFrame,
-                     df_strata: DataFrame) -> tuple[DataFrame, dict[int, str]]:
-    """Get slide-level results."""
-    df = merge(df_assignments, df_strata, on='stratum identifier', how='left')[
-        ['specimen', 'subject diagnosed result']].rename(
-        {'specimen': 'slide', 'subject diagnosed result': 'result'}, axis=1)
-    label_to_result = dict(enumerate(sort(df['result'].unique())))
-    return df.replace({res: i for i, res in label_to_result.items()}), label_to_result
-
-
-def retrieve_importances() -> DataFrame:
-    filename = join('out', 'importances.csv')
-    return read_csv(filename, index_col=0)
-
-
-def save_importances(_args):
-    df = retrieve_importances()
-    credentials = DBCredentials(_args.dbname, _args.host, _args.user, _args.password)
+def save_importance(dbname: str, host: str, user: str, password: str) -> None:
+    """Save cell importance scores as defined by cggnn to the database."""
+    df = read_csv(join('out', 'importances.csv'), index_col=0)
+    credentials = DBCredentials(dbname, host, user, password)
```

Comment:
Can this be a `DBConfig` instead of a 4-tuple input to be consistent with the run and extract methods?

Reply:
I assume you mean `DBCredentials`, but yes. Why not? Is there something I'm missing?

Reply:
Looking back at this, this would actually error in the version committed to `main`. `args.dbname, args.host, args.user, args.password` are provided to the `save_importance` function in the script section of this file, but the `ArgumentParser` only asks for `args.spt_db_config_location` (as does `cggnn run extract`).

Reply:
I see, I didn't catch that.

Reply:
Pushing a commit to fix this...